includes/mutation/class-order-delete.php

```diff
@@ -0,0 +1,151 @@
+<?php
+/**
+ * Mutation - deleteOrder
+ *
+ * Registers mutation for delete an order.
+ *
+ * @package WPGraphQL\Extensions\WooCommerce\Mutation
+ * @since 0.2.0
+ */
+
+namespace WPGraphQL\Extensions\WooCommerce\Mutation;
+
+use GraphQL\Error\UserError;
+use GraphQL\Type\Definition\ResolveInfo;
+use GraphQLRelay\Relay;
+use WPGraphQL\AppContext;
+use WPGraphQL\Extensions\WooCommerce\Data\Mutation\Order_Mutation;
+use WPGraphQL\Extensions\WooCommerce\Model\Order;
+
+/**
+ * Class Order_Delete
+ */
+class Order_Delete {
+	/**
+	 * Registers mutation
+	 */
+	public static function register_mutation() {
+		register_graphql_mutation(
+			'deleteOrder',
+			array(
+				'inputFields'         => self::get_input_fields(),
+				'outputFields'        => self::get_output_fields(),
+				'mutateAndGetPayload' => self::mutate_and_get_payload(),
+			)
+		);
+	}
+
+	/**
+	 * Defines the mutation input field configuration
+	 *
+	 * @return array
+	 */
+	public static function get_input_fields() {
+		$input_fields = array_merge(
+			array(
+				'id'          => array(
+					'type'        => 'ID',
+					'description' => __( 'Order global ID', 'wp-graphql-woocommerce' ),
+				),
+				'orderId'     => array(
+					'type'        => 'Int',
+					'description' => __( 'Order WP ID', 'wp-graphql-woocommerce' ),
+				),
+				'forceDelete' => array(
+					'type'        => 'Boolean',
+					'description' => __( 'Delete or simply place in trash.', 'wp-graphql-woocommerce' ),
+				),
+			)
+		);
+
+		return $input_fields;
+	}
+
+	/**
+	 * Defines the mutation output field configuration
+	 *
+	 * @return array
+	 */
+	public static function get_output_fields() {
+		return array(
+			'order' => array(
+				'type'    => 'Order',
+				'resolve' => function( $payload ) {
+					return $payload['order'];
+				},
+			),
+		);
+	}
+
+	/**
+	 * Defines the mutation data modification closure.
+	 *
+	 * @return callable
+	 */
+	public static function mutate_and_get_payload() {
+		return function( $input, AppContext $context, ResolveInfo $info ) {
+			$post_type_object = get_post_type_object( 'shop_order' );
+
+			if ( ! current_user_can( $post_type_object->cap->create_posts ) ) {
```

Comment:
What if I want to filter and allow or disallow a person to perform this mutation? Maybe add a filter here and on the other mutations?!

Reply:
Fixed.